### PR TITLE
Fix Editor crashes on URDF prefab overwite and import non-valid URDF

### PR DESCRIPTION
**Pull request by michalpelka**

### CODE REVIEW

**Comment by michalpelka** [comment]
That PR fixes: #164 #171, but sadly do not fix #169.

File: Code/Source/RobotImporter/URDF/RobotImporter.cpp [hunk 1]
```diff
@@ -14,16 +14,19 @@ namespace ROS2
     RobotImporter::RobotImporter(LoggerFunction logger)
         : m_logger(std::move(logger))
         , m_isProcessingAssets(false)
+        , m_LoadingURDFFailed(false)
```
Comment (pijaro) [inline Code/Source/RobotImporter/URDF/RobotImporter.cpp]: nitpick
```suggestion
        , m_loadingURDFFailed(false)
```

File: Code/Source/RobotImporter/URDF/RobotImporter.cpp [hunk 2]
```diff
@@ -42,21 +45,28 @@ namespace ROS2
...
+        else if (m_LoadingURDFFailed)
         {
-            auto errorMessage = AZStd::string::format("Importing robot definition failed with error: %s", outcome.GetError().c_str());
-            m_logger(LogLevel::Error, errorMessage);
             importFinishedCb();
             return;
         }
-        m_logger(LogLevel::Info, AZStd::string::format("Imported %s", m_prefabMaker->GetPrefabPath().c_str()));
-        importFinishedCb();
+        AZ_Assert(m_prefabMaker, "Prefab maker is not initialized");
```
Comment (pijaro) [inline Code/Source/RobotImporter/URDF/RobotImporter.cpp]: I believe this is an unnecessary assert. Move it to the top or remove it.

Comment (michalpelka) [reply Code/Source/RobotImporter/URDF/RobotImporter.cpp]: I do not agree - we need to assert when URDF is not failed, but m_prefab optional is empty. I changed simplified logic in this function to make this assert more relevant. Hope you like it more now.

Comment (pijaro) [reply Code/Source/RobotImporter/URDF/RobotImporter.cpp]: I was thinking about the rule in general, that we should assert that "something" is "correct" before we start using it in any way. 

But yes, I see the reason behind that implementation. In this case, I'm ok with it.

**Review by adamdbrw — APPROVED** [review]

**Pull request merged**